### PR TITLE
added instance method isfinite to pytorch tensor (pytorch frontend)

### LINKED ISSUE #21943: isfinite

### PR DESCRIPTION
**Pull request by VeraChristina**
Hi, I have added the instance method isfinite.
Close #21943 
Would you please have a look? Thanks!

### CODE REVIEW

**Comment by VeraChristina** [comment]
I am not able to link the corresponding issue under development. It is: isfinite #21943

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by umairjavaid** [comment]
The tests are not passing. Please fix this and request a review
```
E   ivy.utils.exceptions.IvyBackendException: numpy: nested_map: numpy: nested_map: can't convert np.ndarray of type bfloat16. The only supported types are: float64, float32, float16, complex64, complex128, int64, int32, int16, int8, uint8, and bool.
E   Falsifying example: test_torch_tensor_isfinite(
E       frontend='torch',
E       on_device='cpu',
E       backend_fw='torch',
E       dtype_and_x=(['bfloat16'], [array(-1, dtype=bfloat16)]),
E       init_flags=FrontendMethodTestFlags(
E           num_positional_args=0,
E           as_variable=[False],
E           native_arrays=[False],
E       ),
E       method_flags=FrontendMethodTestFlags(
E           num_positional_args=0,
E           as_variable=[False],
E           native_arrays=[False],
E       ),
E       frontend_method_data=FrontendMethodData(ivy_init_module='ivy.functional.frontends.torch', framework_init_module='torch', init_name='tensor', method_name='isfinite'),
E   )
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.82.4', b'AXicY2YAAkYGCIDRDAAAcgAG') as a decorator on your test case
```

**Comment by VeraChristina** [comment]
The test test_torch_tensor_isfinite now passes (when run with pytest in my environment), there remain unsuccessful checks, but the error messages are not very telling. Where should I look instead?

**Comment by VeraChristina** [comment]
@umairjavaid Can't request a review in the sidebar -- it just says 'No reviews'. Can you please help with the above question?

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
         self.ivy_array = self.fix().ivy_array
         return self
 
+    @with_unsupported_dtypes(
+        {
+            "2.0.1 and below": (
+                "bfloat16",
+                "bool",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1098]: bool is also supported

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
...
+                "uint8",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1099]: uint8 is also supported

File: ivy/functional/frontends/torch/tensor.py [hunk 3]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
...
+                "int8",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1100]: int8 is also supported

File: ivy/functional/frontends/torch/tensor.py [hunk 4]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
...
+                "int16",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1101]: int16 is also supported

File: ivy/functional/frontends/torch/tensor.py [hunk 5]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
...
+                "complex64",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1102]: complex64 is also supported

File: ivy/functional/frontends/torch/tensor.py [hunk 6]
```diff
@@ -1091,6 +1091,23 @@ def fix_(self):
...
+                "complex128",
```
Comment (umairjavaid) [inline ivy/functional/frontends/torch/tensor.py:1103]: complex128 is also supported

File: ivy/functional/frontends/torch/tensor.py (hunk 4 above)
Comment (VeraChristina) [reply ivy/functional/frontends/torch/tensor.py:1101]: But all those tests fail at a deeper level than this instance method.  What's the reason for that?

Comment (VeraChristina) [reply ivy/functional/frontends/torch/tensor.py:1101]: It seems that they are not supported in certain backends? How do I change that?

**Review by umairjavaid — CHANGES_REQUESTED** [review]
Please remove the supported dtype from the @with_unsupported_dtypes

**Comment by VeraChristina** [comment]
I have removed the supported dtypes as requested. Now the tests fail.
I have also run the test test_torch_isfinite for the respective function isfinite in the torch frontend (i.e. the one that my instance method calls) and that test fails (with the same error messages as my test does), so apparently this function does not support these dtypes. Or am I getting this totally wrong? I'd be grateful for feedback.

**Comment by VeraChristina** [comment]
Just for comparison, I ran the tests for the very similar instance method isinf -- same test failures. Please let me know what to do about this PR.

**Comment by umairjavaid** [comment]
You need add bool to the unsupported dtype decorator of the isfinite tensorflow backend function, so that bool is not passed to it during testing. Similarly, you also need to implement isfinite in paddle if it's missing along with its unsupported dtype decorator.

**Comment by VeraChristina** [comment]
Thank you! This was very helpful. I have added bool to unsupp dtypes in tensorflow backend. The function isfinite is already implemented along with the unsupported dtype decorator (which looks correct to me) and I don't know why it's not found during test. Similarly, I don't know why the test for the torch backend fails; as far as I can see bfloat16 should be supported.

**Pull request merged**